src/sinks/azure_blob.rs

```diff
@@ -0,0 +1,817 @@
+        container = ?container_name,
+        blob = ?blob
+    );
+
+    AzureBlobSinkRequest {
+        container_name,
+        blob_data: inner,
+        blob_name: blob,
+        content_encoding: compression.content_encoding(),
+        content_type: compression.content_type(),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::event::LogEvent;
+    use std::collections::BTreeMap;
+
+    #[test]
+    fn generate_config() {
+        crate::test_util::test_generate_config::<AzureBlobSinkConfig>();
+    }
+
+    #[test]
+    fn azure_blob_encode_event_text() {
+        let message = String::from("hello world");
+        let log = LogEvent::from(message.clone());
+        let blob_prefix = Template::try_from("logs/blob/%F").unwrap();
+        let encoding = EncodingConfig {
+            codec: Encoding::Text,
+            schema: None,
+            only_fields: None,
+            except_fields: None,
+            timestamp_format: None,
+        };
+
+        let bytes = encode_event(log.into(), &blob_prefix, &encoding).unwrap();
+
+        let encoded_message = message + "\n";
+        let (bytes, _) = bytes.item.into_parts();
+        assert_eq!(&bytes[..], encoded_message.as_bytes());
+    }
+
+    #[test]
+    fn azure_blob_encode_event_json() {
+        let message = String::from("hello world");
+        let mut log = LogEvent::from(message.clone());
+        log.insert("key", "value");
+        let blob_prefix = Template::try_from("logs/blob/%F").unwrap();
+        let encoding = EncodingConfig {
+            codec: Encoding::Ndjson,
+            schema: None,
+            only_fields: None,
+            except_fields: None,
+            timestamp_format: None,
+        };
+
+        let bytes = encode_event(log.into(), &blob_prefix, &encoding).unwrap();
+
+        let (bytes, _) = bytes.item.into_parts();
+        let map: BTreeMap<String, String> = serde_json::from_slice(&bytes[..]).unwrap();
+        assert_eq!(map[&log_schema().message_key().to_string()], message);
+        assert_eq!(map["key"], "value".to_string());
+    }
+
+    #[test]
+    fn azure_blob_encode_event_with_removed_key() {
+        let message = "hello world".to_string();
+        let mut log = LogEvent::from(message.clone());
+        log.insert("key", "value");
+        let blob_prefix = Template::try_from("logs/blob/%F").unwrap();
+        let encoding = EncodingConfig {
+            codec: Encoding::Ndjson,
+            schema: None,
+            only_fields: None,
+            except_fields: Some(vec!["key".into()]),
+            timestamp_format: None,
+        };
+
+        let bytes = encode_event(log.into(), &blob_prefix, &encoding).unwrap();
+
+        let (bytes, _) = bytes.item.into_parts();
+        let map: BTreeMap<String, String> = serde_json::from_slice(&bytes[..]).unwrap();
+        assert_eq!(map[&log_schema().message_key().to_string()], message);
+        assert_eq!(map.contains_key("key"), false);
```

Comment:
New clippy lints will reject `assert_eq` comparisons to `true` or `false`.
```suggestion
        assert!(!map.contains_key("key"));
```